homeassistant/components/yale_smart_alarm/lock.py

```diff
@@ -0,0 +1,107 @@
+        COORDINATOR
+    ]
+    codeformat = entry.options.get(CONF_LOCK_CODE_DIGITS, DEFAULT_LOCK_CODE_DIGITS)
+
+    async_add_entities(
+        YaleDoorlock(coordinator, data, codeformat)
+        for data in coordinator.data["locks"]
+    )
+
+
+class YaleDoorlock(CoordinatorEntity, LockEntity):
+    """Representation of a Yale doorlock."""
+
+    def __init__(
+        self, coordinator: YaleDataUpdateCoordinator, data: dict, codeformat: int
+    ) -> None:
+        """Initialize the Yale Lock Device."""
+        super().__init__(coordinator)
+        self._coordinator = coordinator
+        self._attr_name = data["name"]
+        self._attr_unique_id = data["address"]
+        self._data = data
+        self._attr_device_info = DeviceInfo(
+            name=self._attr_name,
+            manufacturer=MANUFACTURER,
+            model=MODEL,
+            identifiers={(DOMAIN, data["address"])},
+            via_device=(DOMAIN, CONF_USERNAME),
+        )
+        self._attr_code_format = f"^\\d{codeformat}$"
+
+    async def async_unlock(self, **kwargs) -> None:
+        """Send unlock command."""
+        code = kwargs.get(ATTR_CODE, self._coordinator.entry.options.get(CONF_CODE))
+
+        if code and self._coordinator.yale:
+            get_lock = await self.hass.async_add_executor_job(
+                self._coordinator.yale.lock_api.get, self._attr_name
+            )
+            lock_state = await self.hass.async_add_executor_job(
+                self._coordinator.yale.lock_api.open_lock,
+                get_lock,
+                code,
+            )
+            if lock_state:
+                self._attr_is_locked = False
+                self.async_write_ha_state()
+            LOGGER.debug("Door unlock: %s", lock_state)
+
+    async def async_lock(self, **kwargs) -> None:
+        """Send lock command."""
+
+        if self._coordinator.yale:
+            get_lock = await self.hass.async_add_executor_job(
+                self._coordinator.yale.lock_api.get, self._attr_name
+            )
+            lock_state = await self.hass.async_add_executor_job(
+                self._coordinator.yale.lock_api.close_lock,
+                get_lock,
+            )
+            if lock_state:
+                self._attr_is_locked = True
+                self.async_write_ha_state()
+            LOGGER.debug("Door unlock: %s", lock_state)
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        for lock in self.coordinator.data["locks"]:
+            if lock["address"] == self._attr_unique_id:
+                self._data = lock
+
+        if self._data["_state"] == "locked" and (
+            self._data["_state2"] == "closed" or self._data["_state2"] == "unknown"
+        ):
+            self._attr_is_locked = True
+        LOGGER.debug("Full data %s", self._data)
+        self.async_write_ha_state()
```

Comment:
```suggestion
        super()._handle_coordinator_update()
```

Reply:
Doesn't work. It doesn't update anymore then (no idea why).
But I changed approach and removed this block completely.